packages/dashboard-hardhat-plugin/src/index.ts

```diff
@@ -1,61 +1,191 @@
 
-    console.log("Preparing Truffle compilation");
+  return {
+    host,
+    port
+  };
+}
 
-    // Prepare Truffle compilations
-    let compilations = await FromHardhat.prepareCompilations();
+function getTruffleDashboardUserConfig(
+  config: HardhatConfig,
+  userConfig: HardhatUserConfig
+) {
+  const {
+    dashboardNetworkName = defaults.networkName,
+    dashboardNetworkConfig: dashboardNetworkUserConfig = {}
+  } = userConfig.truffle || {};
 
-    // Create Message Bus client
-    let messageBusClient = new DashboardMessageBusClient(dashboardConfig);
+  const {
+    gas = defaults.gas,
+    gasPrice = defaults.gasPrice,
+    gasMultiplier = defaults.gasMultiplier,
+    timeout = defaults.timeout,
+    httpHeaders = defaults.httpHeaders
+  } = dashboardNetworkUserConfig;
 
-    console.log("Sending Truffle compilation to dashboard");
+  const dashboardNetworkConfig: Pick<
+    HttpNetworkConfig,
+    TruffleDashboardNetworkConfigurableKeys
+  > = {
+    gas,
+    gasPrice,
+    gasMultiplier,
+    timeout,
+    httpHeaders
+  };
 
-    // Send Truffle compilations to dashboard
-    const publishLifecycle = await messageBusClient.publish({
-      type: "cli-event",
-      payload: {
-        label: "workflow-compile-result",
-        data: { compilations }
-      }
-    });
-    publishLifecycle.abandon();
+  const networkIsDefinedExplicitly =
+    userConfig.networks && dashboardNetworkName in userConfig.networks;
 
-    console.log("Compilation successfully sent!");
-  } catch (hardhatError) {
-    console.warn(
-      "The Truffle Dashboard plugin failed to compile: ",
-      hardhatError
+  if (networkIsDefinedExplicitly) {
+    throw new HardhatPluginError(
+      pluginName,
+      `Manual network config disallowed.\n\n` +
+        `This plugin manages your Truffle Dashboard network config for you,\n` +
+        `but your Hardhat config contains \`config.networks["${dashboardNetworkName}"]\`.\n\n` +
+        `Please remove this network and use \`config.truffle.dashboardNetworkConfig\`\n` +
+        `to override any particular network settings.\n\n` +
+        `For example, here's how your config might look with populated default values:\n` +
+        `    module.exports = {\n` +
+        `      networks: {\n` +
+        `        // ... networks config ...\n` +
+        `        // MUST NOT INCLUDE "${dashboardNetworkName}"\n` +
+        `      },\n` +
+        `\n` +
+        `      truffle: {\n` +
+        `        dashboardNetworkName: "${dashboardNetworkName}",\n` +
+        `        dashboardNetworkConfig: {\n` +
+        Object.entries(dashboardNetworkConfig)
+          .map(([name, value]) => `          ${name}: ${util.inspect(value)}`)
+          .join(",\n") +
+        `\n` +
+        `        }\n` +
+        `      }\n` +
+        `    }\n\n` +
+        `Please see the README for more details about how to configure this plugin:\n` +
+        `  https://www.npmjs.com/package/@truffle/dashboard-hardhat-plugin`
```

Comment:
```suggestion
      `Manual network config disallowed.\n\n` +
        `This plugin manages your Truffle Dashboard network config for you,\n` +
        `but your Hardhat config also contains \`config.networks["${dashboardNetworkName}"]\`.\n\n` +
        `You can fix this error by removing \`config.networks["${dashboardNetworkName}"]\`.\n\n`  +
        `Please see the README for more details about how to configure this plugin:\n` +
        `  https://www.npmjs.com/package/@truffle/dashboard-hardhat-plugin`
```

Reply:
You think that the error message shouldn't include the example configuration with all the defaults?